src/SteadyHeap.h

```diff
@@ -0,0 +1,107 @@
+#ifndef _STEADY_HEAP_H_
+#define _STEADY_HEAP_H_
+
+#include <algorithm>
+#include <cassert>
+#include <vector>
+
+namespace pipert {
+
+/// A SteadyHeap wraps the functionality of a heap that can be restrained to
+/// occupy a fix-sized space in memory. It does this by encapsulating a vector
+/// and by providing only the interface to reserve the maximum amount of memory
+/// this heap is allowed to consume, as well as providing methods for handling
+/// its contents as a heap. This container wrapper is specially adapted to our
+/// needs, and contains only the functions the library uses. It does not give a
+/// complete interface for a generic heap, and the implementation is expected to
+/// be extended when new requirements arise in the library.
+///
+/// \tparam T The type of data the heap will contain as its elements.
+/// \tparam StrictOrdering The type of the comparison operation that will be
+/// used to determine the positioning of elements in the heap. It should satisfy
+/// the requirements of _Compare_, meaning that the return value of the
+/// comparison function, when contextually converted to `bool`, should yield
+/// `true` if its first argument appears before the second in the strict weak
+/// ordering induced by the type of its parameters. Comparison using less-than
+/// yields a max heap, and greater-than yields a min heap.
+template <typename T, typename StrictOrdering>
+class SteadyHeap {
```

Comment:
This is perfect. However, I think we could have some very basic tests for this class.

Reply:
Thank you. I wrote some simple tests for it that test the core and custom functionality of the heap, which I will publish soon.